commands/trivia.js

```diff
@@ -260,135 +254,189 @@ module.exports = {
+
+			//We have a winner!
 			if(winner !== null){
-				console.info('logGame post if ' + winner.username);
+				console.info('lo ' + winner.username);
 				winnerID = winner.id;
 			}
-			const game = await Games.create({
+			//Create an entry in the database for this game. Log asyncronously when result is returned. 
+			Games.create({
 				game_id: message.id,
 				creator_id: message.author.id,
 				creator_name: message.author.username,
 				game_start: message.createdAt,
 				game_end: Date.now(),
 				winner_id: winnerID,
 				server_id: message.guild.id,
-			});
+			}).then(value => console.info('Game ' + value.game_id + ' was created in the database'));
 		}
 		
 
 
-		async function logResponse(isWinner, points, user, message, round, reaction, questionTime, answerTime, questionId) {
+		async function logResponse(isWinner, points, user, message, round, questionTime, answerTime, questionId) {
 			console.info('logResponse question: ' + questionId);
-			const userObj = await Users.findOne({ where:
-				{
-					user_id: user.id
-				}});
-			if (userObj === null) {
-				//first time user on this bot
-				logUser(msg, user);
-			} else {
-				console.info('User already resides on the server')
-			}
+			
+			//if the user received points (or isWinner) they got the answer right...
+			// TODO do we need to log correct answer if we make that assumption? 
+			const correctAnswer = points > 0;
+
+			//how to find a user in the db
+			let userSearchCriteria = { where: {
+				user_id: user.id
+			}};
+
+			//if this is the user's first time on the bot then log them. 
+			// TODO could look into checking if this is the first time on the server / if they are on other servers
+			Users.findOne(userSearchCriteria).then(value => {
+				if (value === null) {
+					//first time user on this bot
+					logUser(msg, user);
+				} else {
+					console.info('User already resides on the server')
+				}
+			});
 
-			const response = await Responses.findOne({ where: 
-				{
-					user_id: user.id,
-					game_id: message.id,
-					round_number: round
+			// Combination of userid, gameid and round number should be unique
+			let responseSearchCriteria = { where: {
+				user_id: user.id,
+				game_id: message.id,
+				round_number: round
+			}};
+			
+			//Create a database entry for this users first answer
+			Responses.findOne(responseSearchCriteria).then(value => {
+				if (value === null) {
+					console.info('not found logging response for ' + user.username);
+					Responses.create({
+						game_id: message.id,
+						user_id: user.id,
+						round_number: round,
+						question_id: questionId,
+						   q_time: questionTime,
+						   a_time: answerTime,
+						   correct: correctAnswer,
+						   points: points,
+						   winner: isWinner,
+					}).then(value => console.info('Created DB entry for user ' + value.user_id + ' for round ' + value.round_number));
+				} else {
+					console.info(user.username + ' has already answered');
 				}
```

Comment:
Part of the purpose of the `await` concept is that you can flatten statements like this.  So you could say:

```js
let existingItem = await Responses.findOne (responseSearchCriteria);

if (existingItem) {
    itemToCreate= { /* ... everything in the dictionary inside your "Responses.create( {...} )" call ... */ };
    let newItem = Responses.create (itemToCreate);
    console.info ("Created DB entry for user " + newItem.user_id + "...");
} else {
    ...your existing error printout...
}
```

...which is easier to read and debug.

What you wrote is correct and legal; it's just harder to follow.

Reply:
one of the things I did above was get rid of the awaits as I don't actually need to wait on any of the db functions. 

i could do 

let existingItemPromise = Responses.findOne (responseSearchCriteria);
existingItemPromise.then(return existingItem);

if (existingItem) {
    itemToCreate= { /* ... everything in the dictionary inside your "Responses.create( {...} )" call ... */ };
    let newItem = Responses.create (itemToCreate);
    console.info ("Created DB entry for user " + newItem.user_id + "...");
} else {
    ...your existing error printout...
}

Reply:
hm. where are findOne() and create() defined? If they're defined with `async` (meaning, someone literally typed that word before the name of the function), and/or the `return` statement in those functions returns an object called `Promise` instead of the content you actually want, then you DO have to use "await."

The code in your comment above won't work, as far as I know:
- line 1 does indeed retrieve the Promise, if in fact `findOne()` was declared that way
- line 2 does nothing: it executes instantly, and doesn't queue up anything to be done later
- line 3 runs immediately, instantly, after line 2. So if findOne() was declared as "async", findOne() hasn't run yet. Therefore existingItem doesn't exist yet. Just as importantly, the code as written doesn't actually declare `existingItem` anywhere, so line 3 will always "fail", because the value is literally `undefined`.

The "await" concept means:  the code will stop at that point, and wait for some time-consuming thing to happen.  Without the "await," by the time your function above executes, everything on the left-hand side will execute exactly as written, in sequence, without waiting for anything out there in the world (such as a database retrieval) to happen, and you won't get your data.